ada/extensions/nodes/ada_node/bodies

```diff
@@ -35,3 +35,147 @@ begin
       Reparse  => False,
       Rule     => Default_Grammar_Rule);
 end P_Standard_Unit;
+
+-----------------------------
+-- P_Filter_Is_Imported_By --
+-----------------------------
+
+function P_Filter_Is_Imported_By
+  (Node       : access Bare_Ada_Node_Type'Class;
+   Units      : Internal_Unit_Array_Access;
+   Transitive : Boolean) return Internal_Unit_Array_Access is
+
+   package Analysis_Unit_Vectors is new Ada.Containers.Vectors
+     (Index_Type   => Positive,
+      Element_Type => Internal_Unit);
+
+   package Analysis_Unit_Maps is new Ada.Containers.Hashed_Maps
+     (Key_Type        => Internal_Unit,
+      Element_Type    => Boolean,
+      Hash            => Hash,
+      Equivalent_Keys => "=");
+
+   Target              : Internal_Unit := Node.Unit;
+   Units_Import_Target : Analysis_Unit_Maps.Map;
+
+   function Does_Import_Target (From : in Internal_Unit) return Boolean;
+   --  Predicate that returns True iff the given unit imports the target.
+   --  If Transitive is True, handle transitive imports.
+
+   function Does_Import_Target (From : in Internal_Unit) return Boolean is
+      function Handle_Unit_Name
+        (Symbols : in Libadalang.Env_Hooks.Symbol_Type_Array)
+         return Boolean;
+      --  Fetches the unit associated to the given name and returns whether
+      --  this unit IS the target unit. If Transitive is True, recursively look
+      --  if the target is one of the units imported by the given unit.
+
+      function Handle_Unit_Name
+        (Symbols : in Libadalang.Env_Hooks.Symbol_Type_Array)
+         return Boolean is
+
+         Unit : Internal_Unit := Libadalang.Env_Hooks.Fetch_Unit
+           (Node.Unit.Context, Symbols, Node.Unit,
+            Unit_Specification, True, False);
+      begin
+         if Unit = Target or else
+              (Transitive and then
+               Does_Import_Target (Unit)) then
+            return True;
+         end if;
+         return False;
+      exception
+         when Property_Error =>
+            return False;
+      end Handle_Unit_Name;
+
+      AST     : Bare_Compilation_Unit := Bare_Compilation_Unit (Root (From));
+      Prelude : Bare_Ada_Node_List;
+
+      Memoized_Position : Analysis_Unit_Maps.Cursor :=
```

Comment:
I would suggest to rename this to “From_Cursor”. It’s more descriptive as to what it contains, and I don’t think “memoized” helps understanding its role (the notion of memoized position itself is debatable IMO!)